File: practica3/python/cola.py
class Cola:
    def __init__(self, n):
        self.arreglo = [0] * n
        self.inicio = 0
        self.fin = -1
        self.n = n

    def insert(self, e):
        if self.isFull():
            print("Cola llena")
        else:
            self.fin += 1
            self.arreglo[self.fin] = e

    def remove(self):
        if self.isEmpty():
            print("Cola vacía")
            return -1
        else:
            e = self.arreglo[self.inicio]
            self.inicio += 1
            return e

    def peek(self):
        if self.isEmpty():
            print("Cola vacía")
            return -1
        return self.arreglo[self.inicio]

    def size(self):
        return self.fin - self.inicio + 1

    def isEmpty(self):
        return self.inicio > self.fin

    def isFull(self):
        return self.fin == self.n - 1

if __name__ == "__main__":
    cola = Cola(3)
    cola.insert(10)
    cola.insert(20)
    cola.insert(30)

    print(cola.peek())
    print("remove() =", cola.remove())
    print("remove() =", cola.remove())
    print("remove() =", cola.remove())

   

File: practica3/Java/Cola.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package cola;

/**
 *
 * @author Administrador
 */
public class Cola {
    public int arreglo[];
    public int inicio, fin;
    public int n;

    public Cola(int n) {
        arreglo = new int[n];
        inicio = 0;
        fin = -1;
        this.n = n;
    }

    public void insert(int e) {
        if (isFull()) {
            System.out.println("Cola llena");
        } else {
            fin++;
            arreglo[fin] = e;
        }
    }

    public int remove() {
        if (isEmpty()) {
            System.out.println("Cola vacía");
            return -1;
        } else {
            int e = arreglo[inicio];
            inicio++;
            return e;
        }
    }

    public int peek() {
        if (isEmpty()) {
            System.out.println("Cola vacía");
            return -1;
        }
        return arreglo[inicio];
    }

    private boolean isEmpty() {
        return (inicio > fin);  
    }

    private boolean isFull() {
        return (fin == n - 1);  
    }
     public int size() {
        return fin - inicio + 1;
    }
    public static void main(String[] args) {
        Cola cola = new Cola(3);
        cola.insert(10);
        cola.insert(20);
        cola.insert(30);

        System.out.println(cola.peek()); 
        System.out.println("remove() = " + cola.remove());  
        System.out.println("remove() = " + cola.remove()); 
        System.out.println("remove() = " + cola.remove()); 

    }
}


